### PR TITLE
Add burstH2MM

### PR DESCRIPTION
**Pull request by harripd**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by harripd** [comment]
I am willing to be a maintainer.

**Comment by harripd** [comment]
@conda-forge/help-python ready for review/

File: recipes/bursth2mm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+requirements:
+  host:
+    - python >=3.8
+    - setuptools >=64
+    - setuptools-scm >=8.0
+    - pip
+  run:
+    - python >=3.8
+    - numpy
+    - matplotlib-base
```
Comment (ocefpaf) [inline recipes/bursth2mm/meta.yaml:29]:
```suggestion
    - matplotlib-base >=3.0.1
```

File: recipes/bursth2mm/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    - SETUPTOOLS_SCM_PRETEND_VERSION={{ version }}
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
+  number: 0
+
+requirements:
+  host:
+    - python >=3.8
+    - setuptools >=64
+    - setuptools-scm >=8.0
+    - pip
```
Comment (ocefpaf) [inline recipes/bursth2mm/meta.yaml:25]: While I could not find any numpy compiled extension upstream, it does declare a build dependency on numpy in https://github.com/harripd/burstH2MM/blob/670a88b4ae033fa6c2a4bc499be003f5ca62e026/pyproject.toml#L6

Maybe that is a mistake b/c the recipe seems to work fine as noarch here.

File: recipes/bursth2mm/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+{% set name = "burstH2MM" %}
+{% set version = "0.1.8" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/bursth2mm/meta.yaml:9]:
```suggestion
```

File: recipes/bursth2mm/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  url: https://github.com/harripd/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: d06654d8e37a1d30ce39c97f861bcc1542b25333fecf43ae5c1a07a2e1892a8f
+
+build:
+  noarch: python
+  script_env:
+    - SETUPTOOLS_SCM_PRETEND_VERSION={{ version }}
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
```
Comment (ocefpaf) [inline recipes/bursth2mm/meta.yaml:17]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

File: recipes/bursth2mm/meta.yaml (hunk 2 above)
Comment (harripd) [reply recipes/bursth2mm/meta.yaml:25]: Both FRETBursts and H2MM_C are dependent on numpy, in pip it uses oldest-supported-numpy, which is then replaced with pin-compatible-numpy in their conda-forge recipes

File: recipes/bursth2mm/meta.yaml (hunk 3 above)
Comment (harripd) [reply recipes/bursth2mm/meta.yaml:9]: Fine with me

File: recipes/bursth2mm/meta.yaml (hunk 1 above)
Comment (harripd) [reply recipes/bursth2mm/meta.yaml:28]: Fine with me

File: recipes/bursth2mm/meta.yaml (hunk 4 above)
Comment (harripd) [reply recipes/bursth2mm/meta.yaml:16]: As long as it still builds

File: recipes/bursth2mm/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipes/bursth2mm/meta.yaml:24]: > Both FRETBursts and H2MM_C are dependent on numpy

To run or build? B/c you don't have it here in host, which is used to build and, if that is true, it should fail as noarch, but it didn't. oldest-supported-numpy/pin-compatible-numpy are used when the package has a compiled extension against numpy, which doesn't seem to be the case, even though it is declared upstream.

File: recipes/bursth2mm/meta.yaml (hunk 4 above)
Comment (ocefpaf) [reply recipes/bursth2mm/meta.yaml]: @harripd, I'm a conda-forge core member and staged-recipes reviewer. Why do you think I would send you something that would not work?

File: recipes/bursth2mm/meta.yaml (hunk 2 above)
Comment (harripd) [reply recipes/bursth2mm/meta.yaml:24]: Not sure why. I included numpy in the build requirements long ago, just assuming it was needed. 

I checked on my local machine, I can remove numpy from the build requirements and it makes no difference. So it does not actually require numpy to build, just to run.

Not sure why it doesn't give an error though given that numpy is stated in the build requirements of the current repo...

File: recipes/bursth2mm/meta.yaml (hunk 4 above)
Comment (harripd) [reply recipes/bursth2mm/meta.yaml]: I didn't see any indication that it couldn't build, I wasn't saying I did trust you, sorry if I implied that.

File: recipes/bursth2mm/meta.yaml (hunk 2 above)
Comment (harripd) [reply recipes/bursth2mm/meta.yaml:24]: Would you like me to make an update (also on pip where I remove the numpy *build* dependency? Just let me know and I'll make the appropriate updates (new version, update the version number and sha256 sum or add pin-compatible-numpy to build) so I can resolve the converstaion

Comment (ocefpaf) [reply recipes/bursth2mm/meta.yaml:24]: This is not a problem for this package now, but auto-updating tools will be confused. It is a bug upstream b/c it declares something that it doesn't need.

**Pull request merged**